convert2rhel/checks.py

```diff
@@ -284,13 +284,20 @@ def ensure_compatibility_of_kmods():
                 unsupported_kmods,
             )
         )
-        logger.critical(
-            (
+        if "CONVERT2RHEL_UNSUPPORTED_UNCHECKED_KMODS" in os.environ:
+            logger.warning(
+                "The following kernel modules are not supported in RHEL:\n{kmods}\n"
+                "'CONVERT2RHEL_UNSUPPORTED_UNCHECKED_KMODS' environment variable detected, continuing conversion."
+            ).format(kmods=not_supported_kmods, system=system_info.name)
```

Comment:
There is one missing part of this string formatting that is missing: 
```suggestion
            logger.warning(
                "The following kernel modules are not supported in RHEL:\n{kmods}\n"
                "'CONVERT2RHEL_UNSUPPORTED_UNCHECKED_KMODS' environment variable detected, continuing conversion.".format(kmods=not_supported_kmods, system=system_info.name)
            )
```

You were trying to call the `format()` method that should come from whatever `warning` returns, but that wouldn't work since `warning()` doesn't return anything.  By moving the `format()` to be at the end of the string, then it will be correct.

Reply:
ahhh, now that you explain it, you are totally right! thank you for teaching me about this!